packages/@lwc/signals/src/index.ts

```diff
@@ -1,19 +1,24 @@
 /*
- * Copyright (c) 2023, salesforce.com, inc.
+ * Copyright (c) 2024, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
+
+import { SIGNAL_SYMBOL } from '@lwc/shared';
+
 export type OnUpdate = () => void;
 export type Unsubscribe = () => void;
 
 export interface Signal<T> {
+    [SIGNAL_SYMBOL]: undefined;
```

Comment:
```suggestion
```
Is this interface intended to describe the generic signal pattern or the LWC implementation, specifically? `SIGNAL_SYMBOL` is a quasi-hidden internal detail, it doesn't need to be part of the public interface. (Downstream code can always safely check `SIGNAL_SYMBOL in someSignal`, even if we don't declare it as part of the interface.)